tools/topology/topology2/include/common/audio_format.conf

```diff
@@ -0,0 +1,237 @@
+# Class definition for audio format object
+# audio_format objects can be instantiated as:
+#
+# Object.Base.audio_format."0" {
+#	in_rate			48000
+#	in_sample_container_size	16
+#	in_valid_bit_depth		16
+#	in_interleaving_style		"interleaved"
+#	out_rate			48000
+#	out_sample_container_size	16
+#	out_valid_bit_depth		16
+#	out_interleaving_style	"	interleaved"
+
+# }
+#
+
+Class.Base."audio_format" {
+
+	DefineAttribute."instance" {
+	}
+
+	# input sampling rate
+	DefineAttribute."in_rate" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	# input bit depth
+	DefineAttribute."in_bit_depth" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	# input valid bit depth
+	DefineAttribute."in_valid_bit_depth" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	# input channel count
+	DefineAttribute."in_channels" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+		constraints {
+			min 1
+			max 8
+		}
+	}
+
+	# input channel map
+	DefineAttribute."in_ch_map" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	# input channel config
+	DefineAttribute."in_ch_cfg" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	# input interleaving style
+	DefineAttribute."in_interleaving_style" {
+		type	"string"
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+		constraints {
+			!valid_values [
+				"interleaved"
+				"non-interleaved"
+			]
+			!tuple_values [
+				0
+				1
+			]
+		}
+	}
+
+	# input format config
+	DefineAttribute."in_fmt_cfg" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	# output sampling rate
+	DefineAttribute."out_rate" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	# input sample_type
+	DefineAttribute."in_sample_type" {
+		type string
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+		constraints {
+			!valid_values [
+				$SAMPLE_TYPE_MSB_INTEGER
+				SAMPLE_TYPE_LSB_INTEGER
+				$SAMPLE_TYPE_SIGNED_INTEGER
+				$SAMPLE_TYPE_UNSIGNED_INTEGER
+				$SAMPLE_TYPE_FLOAT
+			]
+		}
+	}
+
+	# output bit depth
+	DefineAttribute."out_bit_depth" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	# output valid bit depth
+	DefineAttribute."out_valid_bit_depth" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
```

Comment:
we can also add constrains for value of 16, 24, 32. I am not sure about 8bits

Reply:
no way for me to specify valid bits like that. I can add min/max of valid string values. Valid integer values would have to be define. let me think about it